### PR TITLE
[tools] Remove support for `support` filename flag

### LINKED ISSUE #15749: Removing `-support` naming convention
When identifying tests, the WPT infrastructure ignores files whose name (not including extension) ends in `-support`. Today, only 13 files meet that criteria:

<details>
  <summary><code>$ git ls-files | grep '\-support\.'</code></summary>

    WebIDL/ecmascript-binding/constructors-support.html
    css/css-values/vh-calc-support.html
    css/css-values/vh-support.html
    css/css-values/vh-zero-support.html
    dom/traversal/traversal-support.js
    event-timing/resources/event-timing-support.js
    eventsource/eventsource-onmessage-realm-support.htm
    fullscreen/rendering/fullscreen-pseudo-class-support.html
    html/semantics/forms/resetting-a-form/reset-form-event-realm-support.html
    html/syntax/parsing/DOMContentLoaded-defer-support.js
    html/the-xhtml-syntax/parsing-xhtml-documents/xhtml-mathml-dtd-entity-support.htm
    html/webappapis/scripting/events/resources/compiled-event-handler-settings-objects-support.html

</details>

The infrastructure *also* ignores files placed within subdirectories named "resources", "tools", or "support". That approach is far more common:

    $ git ls-files | grep -E '(/|^)(resources|tools|support)/.*\.(html?|xht)' | wc -l
    851

I'd like remove the infrastructure that recognizes the `-support` file name pattern and relocate the files that rely on it to an appropriately-named subdirectory. My motivation is to reduce the learning curve for contribution and limit the number of ways to express the same thing. In addition, four of the `-support` files listed above appear to actually be tests. If so, they are evidence that the naming scheme is prone to accidental opt-in.

Does anyone know of a reason why this simplification cannot or should not be made? @gsnedders or @jgraham, perhaps?

### PR DESCRIPTION
**Pull request by jugglinmike**
In terms of file interpretation by the WPT infrastructure, naming a file
with the `-support` flag is equivalent to placing it in a directory
named `support`, `resources`, or `tools`. The ability to express the
same intention with different mechanisms allows for variation that adds
no value and needless increases the complexity of the infrastructure and
its documentation. Further, the character sequence of `-support` is
prone to accidental opt-in: a number of tests in this suite currently
include that sequence and are thus not executed in automation.

Remove support for the flag.

Resolves gh-15749

### CODE REVIEW

File: eventsource/eventsource-onmessage-realm.htm [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
 
-<iframe src="eventsource-onmessage-realm-support.htm"></iframe>
+<iframe src="resources/eventsource-onmessage-realm.htm"></iframe>
 
 <script>
 "use strict";
 
 async_test(t => {
   window.onload = t.step_func(() => {
-    const source = new frames[0].EventSource("resources/message.py");
```
Comment (annevk) [inline eventsource/eventsource-onmessage-realm.htm]: Did you run this particular change through all browsers? It strikes me they might not use the correct global to parse URLs against.

**Review by annevk — APPROVED** [review]

File: eventsource/eventsource-onmessage-realm.htm (hunk 1 above)
Comment (jugglinmike) [reply eventsource/eventsource-onmessage-realm.htm]: Nice eye! I verified this change in Firefox and Chromium; it does not effect the results reported by either browser for the affected tests. I'm not able to verify in Safari, though.

**Comment by jugglinmike** [comment]
The CI failure is due to an unrelated change in one of our dependencies. I've submitted gh-17648 to address that.

**Comment by Ms2ger** [comment]
It's not clear to me that it's necessary to move these files; if they don't include th.js, they should be considered to be support files by default.

**Comment by annevk** [comment]
Even so, I'd strongly prefer moving them. Having non-test files in a separate directory makes it much clearer when manually analyzing things.

**Comment by jugglinmike** [comment]
@annevk that's my thinking, as well. Although for the record, it's a losing battle. As of last week, WPT had over 7,000 non-test files which are named like tests:

    $ cat ../docs/2019-07-04-files-labeled-support-by-default.txt | grep -E '\.(html|htm|xht|xhtml|svg|js)$' | grep -Eo '[^/.]+$' | sort | uniq -c
         17 htm
       4527 html
        605 js
        197 svg
       2782 xht
        622 xhtml

**Comment by annevk** [comment]
What if you remove all `-ref` tests? And how many of those are old versus new? I hope there might be some nuance here and it's not as bad.

**Comment by jugglinmike** [comment]
You're right. Things do lookg more cheery if we ignore files whose name contain `-ref`:

    $ cat ../docs/2019-07-04-files-labeled-support-by-default.txt | grep -E '\.(html|htm|xht|xhtml|svg|js)$' | grep -v -e '-ref' | grep -Eo '[^/.]+$' | sort | uniq -c 
          1 htm
        977 html
        602 js
         98 svg
         63 xht
        546 xhtml

Still over 2k, though. [I've put the complete list online here](https://gist.github.com/jugglinmike/44f5a2d5a73a6818b20509744009d653) in case you or anyone else wants to take a closer look.

In any case, gh-17648 has been merged, so I'll close and re-open this. (I *think* that will run this patch with the fix...)

**Comment by jugglinmike** [comment]
It's been a week, so I just checked whether any new `-support` files have been added to WPT in that time. No such files have been added, so I'll merge this myself. Thanks for the review!

**Pull request merged**